### PR TITLE
Implement Email-Sending Capability in `email-service` with Kafka Communication  

### PR DESCRIPTION
**Pull request by wade3hamati**
#### **Summary**  
This PR introduces the implementation of an email-sending capability in the `email-service`. The service is designed to listen to Kafka topics for events and send emails based on the event payload. This ensures that notifications, alerts, and other communications can be reliably delivered to users.

#### **Key Changes**  
1. **Kafka Listener**:  
   - Configured `email-service` to subscribe to the `email-sent` Kafka topic.
   - Implemented a Kafka listener to consume and process incoming messages containing email-related data.  

2. **SMTP Configuration**:  
   - Integrated SMTP settings for email delivery using a Resend.  

3. **Email Templates**:  
   - Added support for customizable templates for different email types to be created in the future.

4. **Error Handling and Logging**:  
   - Implemented robust error handling for failed email sends, with detailed logs for diagnostics.  
 
#### **Acceptance Criteria**  
- The `email-service` listens to the `email-notifications` Kafka topic and processes events correctly.  
- Emails are sent using SMTP settings, and content is dynamically customized with templates.  
- Errors are logged, and retry logic handles transient failures gracefully.  

#### **Story Details**  
- **Story Points**: 13  
- **Priority**: P1  
- **Risk**: High  

#### **Next Steps**  
- Monitor `email-service` performance in production for edge cases and optimize as needed.  
- Extend the template library to include more email types as required by the system.  
- Add retry logic for transient SMTP errors (e.g., timeouts).  
- Test the microservice especially as it starts to grow.

### CODE REVIEW

**Comment by walidoow** [comment]
@wade3hamati can you delete the `.idea` directory? Use the `git rm .idea --cached` command.

File: Microservices/email-service/src/main/java/app/sportahub/emailservice/controller/EmailController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import app.sportahub.emailservice.dto.request.EmailRequestDTO;
+import app.sportahub.emailservice.service.EmailService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/api/emails")
```
Comment (walidoow) [inline Microservices/email-service/src/main/java/app/sportahub/emailservice/controller/EmailController.java:11]: Replace the `@RequestMapping("/api/emails")` by `@RequestMapping("/email")` and add this line to the server.properties file of the `email-service`
```yaml
server.servlet.context-path=${SERVER_SERVLET_CONTEXT_PATH:/api/email-service}
```

**Review by walidoow — CHANGES_REQUESTED** [review]
Everything looks good 👌, just make sure to change the things that I asked. Also, don't forget to delete the .idea file from the project. Keep up the good work.

**Review by walidoow — APPROVED** [review]
Everything looks good to me!

**Review by DanDuguay — APPROVED** [review]
Looks great!
